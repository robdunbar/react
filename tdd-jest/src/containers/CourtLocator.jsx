import React, { Component } from 'react';
import Header from '../components/Header';
import Button from '../components/Button';
import Map from '../components/Map';
import MapChooser from '../mapChooser';
import mapChooser from '../mapChooser';
import axios from 'axios';

class CourtLocator extends Component {
  constructor(props)
  {
    super(props);

    this.state = {
      currentMap: '',
      courts: []
    };
    
    this.chooseMap = this.chooseMap.bind(this);
  }

  async componentDidMount(){
    let response = await axios.get('http://localhost:3000/data/courts.json');
    this.setState({
      currentMap: this.state.currentMap,
      courts: response.data.courts
    })
  }

  chooseMap(e){
    const theButton = e.target;
    this.setState({currentMap: mapChooser(theButton.value)});
  }

  render() {
    let courtButtons = this.state.courts.map((court, index) => {
      return <Button {...court} key={index} handleClick={this.chooseMap} />
    });

    return (<div>
              <Header />
              <div >
                {courtButtons}
              </div>

              <Map imageName={this.state.currentMap} location={this.props.location} />
            </div>);
  }
}

export default CourtLocator;
