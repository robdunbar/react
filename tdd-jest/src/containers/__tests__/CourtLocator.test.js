import React from 'react';
import {shallow} from 'enzyme';
import CourtLocator from '../CourtLocator';
import axios from 'axios';
import renderer from 'react-test-renderer';

describe("CourtLocator", function(){

  let mountedCourtLocator;
  beforeEach(()=>{
    mountedCourtLocator = shallow(<CourtLocator />);
  });

  it('renders without crashing', () => {
    mountedCourtLocator = shallow(<CourtLocator />);
  });

  it('renders header', () => {
    var headers = mountedCourtLocator.find('Header');
    expect(headers.length).toBe(1);
  });

  it('renders map', () => {
    var maps = mountedCourtLocator.find('Map');
    expect(maps.length).toBe(1);
  });

  it ('renders correctly', ()=>{
    const tree = renderer.create(<CourtLocator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls axios.get in #componentDidMount', ()=>{
    return mountedCourtLocator.instance().componentDidMount().then(
      ()=>{
        expect(axios.get).toHaveBeenCalled();
      }
    );
  });

  it('calls axios.get with correct url', ()=>{
    return mountedCourtLocator.instance().componentDidMount().then(
      ()=>{
        expect(axios.get).toHaveBeenCalledWith('http://localhost:3000/data/courts.json');
      }
    );
  });

  it('updates state with api data', ()=>{
    return mountedCourtLocator.instance().componentDidMount().then(
      ()=>{
        expect(mountedCourtLocator.state()).toHaveProperty('courts', [
          {
            "location": "test location",
            "address": "test address"
          }
        ]);
      }
    );
  });
});

describe('chooseMap', ()=>{
  it('updates this.state.currentMap using the location passed to it', ()=>{
    const mountedCourtLocator = shallow(<CourtLocator />);
    const mockEvent = {target:{value:'testland'}};
    mountedCourtLocator.instance().chooseMap(mockEvent);
    expect(mountedCourtLocator.instance().state.currentMap).toBe('images/testland.png');
  });
});