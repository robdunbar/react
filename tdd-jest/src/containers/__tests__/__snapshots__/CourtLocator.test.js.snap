// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourtLocator renders correctly 1`] = `
<div>
  <div
    className="Header"
  >
    <img
      alt="Logo"
      src="images/logo.jpg"
      width="480"
    />
    <h1>
      Basketball Court Locator
    </h1>
    <p>
      Powered by React v
      16.6.3
    </p>
  </div>
  <div />
  <div
    className="MapBox"
  >
    <img
      alt="Map displaying the location of basketball courts."
      src="images/None.png"
    />
  </div>
</div>
`;
